src/dev_lua_lib.erl

```diff
@@ -196,4 +200,106 @@ event([Group, Event], ExecState, Opts) ->
         }
     ),
     ?event(Group, Event),
-    {[<<"ok">>], ExecState}.
\ No newline at end of file
+    {[<<"ok">>], ExecState}.
+
+
+-ifdef(ENABLE_LUA_CACHE). 
+
+%% Lua Cache Functions Wrappers
+
+%% @doc Wrapper for hb_cache:write/2 (structured message version).
+%% Expects Lua call: ao.cache_write(message_map, opts?)
+%% Returns [ok, UncommittedID] | [error, #{reason => Reason}]
+cache_write(Args, ExecState, ExecOpts) ->
+    [MsgMapLua | RestArgs] = Args,
+    MsgMapErlang = dev_lua:decode(MsgMapLua),
+    LuaOpts = maybe_decode_opts(RestArgs),
+    FinalOpts = maps:merge(ExecOpts, LuaOpts),
+    ?event(debug_lua_lib_cache, {cache_write_call, {msg, MsgMapErlang}, {final_opts, FinalOpts}}),
+    Result = hb_cache:write(MsgMapErlang, FinalOpts),
+    ReturnList = case Result of
+        {ok, ID} -> [ok, ID]; % Return [ok, ID]
+        {error, Reason} -> [error, #{ <<"reason">> => Reason }] % Return [error, ReasonMap]
+    end,
+    ?event(debug_lua_lib_cache, {cache_write_return_list, {list, ReturnList}}),
+    {ReturnList, ExecState}.
+
+%% @doc Wrapper for hb_cache:read/2.
+%% Expects Lua call: ao.cache_read(id_or_path_string, opts?)
+%% Returns [ok, Value] | [not_found] | [error, #{reason => Reason}]
+cache_read(Args, ExecState, ExecOpts) ->
+    [IDOrPathBin | RestArgs] = Args,
+    LuaOpts = maybe_decode_opts(RestArgs),
+    FinalOpts = maps:merge(ExecOpts, LuaOpts),
+    ?event(dev_lua_lib, {cache_read_call, {id_or_path, IDOrPathBin}, {final_opts, FinalOpts}}),
+    Result = hb_cache:read(IDOrPathBin, FinalOpts),
+    ReturnList = case Result of
```

Comment:
Same here, lists are required for luerl support

Reply:
same, re: your comment from earlier 

```{ok, Value} -> [ok, Value]; % Return [ok, Value]
        not_found -> [not_found]; % Return [not_found]```

is this the list support you meant (eg, not_found is a single list element, [error, <reason>] is another one?